### PR TITLE
Drag and Drop N Days

### PR DESCRIPTION
**Pull request by meganyin13**
### Summary 

Allow drag and drop across n days.

WORK PLAN/ISSUES:
- [x]  allow drag and drop n days in 5D view
- [x]  allow drag and drop to day with no tasks
- [x]  bring back task reordering for drag and drop same day
- [x]  disable dragging ability for repeating task
- [x]  disable dragging ability for repeating task
- [x]  fix 2W and monthly view
- [x]  make dark mode work with drag and drop

This pull request implements the long-awaited feature Drag and Drop (single day + across N days)

- [x] implements Drag and Drop!

### Test Plan 

Make at least 3 tasks on at least 2 different days. For clarity, number each task and label the day the task starts on. Drag these tasks to different dates, ensure they appear where you expect. Also try dragging these tasks to dates with no tasks. Create a repeated task and ensure that you are unable to drag it.

Ensure that single day drag and drop is unaffected by dragging tasks within one day.

### Known Issues
WILL BE FIXED IN NEXT ITERATION:
- Dragging and dropping about a repeated task on a single day causes some wacky UI behavior. 
- Need to use caching to fix the flash that occurs after dropping. 
- We are CORRUPTING the database. FutureView order should be independent of FocusView. #408.

### CODE REVIEW

File: frontend/src/components/TaskView/FutureView/FutureViewDayTaskContainer.module.css [hunk 1]
```diff
@@ -1,3 +1,4 @@
 .Container {
   height: calc(100% - 66px);
+  min-height: 216px;
```
Comment (SamChou19815) [inline frontend/src/components/TaskView/FutureView/FutureViewDayTaskContainer.module.css]: This is problematic. Things might only work for certain screen size.

Comment (ptwu) [reply frontend/src/components/TaskView/FutureView/FutureViewDayTaskContainer.module.css]: we can try to calculate the container height instead, or an easier way to deal with this is to just disable drag and drop for 2W and monthly views

**Comment by mt-xing** [comment]
You're not supposed to be able to change the due date of Canvas tasks, yet you can still drag them into different days.

**Comment by mt-xing** [comment]
Dark mode is broken

**Comment by mt-xing** [comment]
Why do Canvas tasks have a repeating icon?
image

**Comment by mt-xing** [comment]
Whenever I drag a task to a new day, it always shows up at the top of that day, no matter where I drop it.

Also dark mode is still broken.

**Pull request merged**